db/commitlog/commitlog.cc

```diff
@@ -264,6 +264,44 @@ class db::commitlog::segment_manager : public ::enable_shared_from_this<segment_
     using request_controller_units = semaphore_units<timeout_exception_factory, db::timeout_clock>;
     request_controller_type _request_controller;
 
+    class named_file : public seastar::file {
```

Comment:
Since this class is internal to commitlog.cc, I shouldn't hold it to a very high standard of documentation, so it's not a disaster that there is none. I just have two minor comments you can feel free not to do:
1. The name "named_file" suggests that tracking the name is the main goal of this file, while it seems its real raison d'etre is tracking the file's size. So maybe the name "size" should have been part of the class name?
2. The method name "known_size()" suggests to me that maybe it doesn't really track the file's size - it's only what is "known" and maybe there're cases when we don't know? What cases are those? If the size is always tracked correctly, and always known, why not just call it size()?

Reply:
Actually, carrying the name is probably "primary", since we only care about the size iff we are forced to delete things. But naming is ugh. 
It is called "known size" because it is the size we infer/assume based on partially actually reading size (on open) and IO ops (write, truncate, allocate). Actual file size could be foobar:ed by an external agent, but we ignore this and only care about size we created. So `known` is correct, but only guaranteed so from "our" standpoint. 

(also to not collide with the actual `future<uint64_t> size() const` operation).